flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/FlinkConfigOptions.java

```diff
@@ -48,6 +48,8 @@
  */
 public class FlinkConfigOptions {
 
+  public static final int MAX_SNAPSHOTS_PER_MONITOR_INTERVAL_DEFAULT = -1;
```

Comment:
we can default it to Integer.MAX_VALUE. then there is no need to check default value.

Reply:
@stevenzwu , Kyle suggested using -1 here: https://github.com/apache/iceberg/pull/4943#discussion_r889652532. I think that makes sense to me as well.

Reply:
I see the `ScanSummary` also use `Integer.MAX_VALUE`. -1 default seems to be used mostly for invalid ids (like snapshotId, fieldId etc.)
```
    private int limit = Integer.MAX_VALUE;
```

Reply:
@kbendick What do you think for this?

Reply:
I’m ok with INT_MAX if that’s what’s used elsewhere in `ScanSummary`. I suggested -1 as we use -1 to disable cache expiration in the caching catalog and because it’s easier to use in things like Python where INT_MAX is less commonly used. -1 is also easier to use as in-line SQL option imo.

But if INT_MAX is already used in the `ScanSummary`, I suggest we keep that consistent.

We can then later on consider using -1 in both places within Flink (or more places). But consistency is better imo.